ipa-core/src/protocol/ipa_prf/boolean_ops/share_conversion_aby.rs

```diff
@@ -153,18 +165,21 @@ where
     // addition x+rs, where rs=r+s might cause carry
     // this is not a problem since bit 255 of rs is set to 0
     let (sh_y, _) = integer_add::<_, TwoHundredFiftySixBitOpStep, NC>(
-        ctx.narrow(&Step::IntegerAddMaskToX),
+        m_ctx.narrow(&Step::IntegerAddMaskToX),
         record_id,
         &sh_rs,
         &x,
     )
     .await?;
 
+    // validate before reveal
+    validator.validate(0).await?;
```

Comment:
I really don't like this passing in 0 as an argument. I'm not a fan of this API design.